components/concept/engine/src/main/java/mozilla/components/concept/engine/prompt/PromptRequest.kt

```diff
@@ -63,4 +66,20 @@ sealed class PromptRequest {
         val onSelect: (java.util.Date) -> Unit,
         val onClear: () -> Unit
     ) : PromptRequest()
+
+    /**
+     * Value type that represents a request for a selecting one or multiple files.
+     * @property mimeTypes a set of allowed mime types. Only these file types can be selected.
+     * @property isMultipleFilesSelection true if the user can select more that one file false otherwise.
+     * @property onSingleFileSelected callback to notify that the user has selected a single file.
+     * @property onMultipleFilesSelected callback to notify that the user has selected multiple files.
+     * @property onDismiss callback to notify that the user has canceled the file selection.
+     */
+    data class FilePicker(
```

Comment:
nit: I'd prefer just `File` in line with the others `Date`, `MenuChoice`, or alternatively we could spell it out `FilePromptRequest`.